keycastr/SvelteVisualizer.m

```diff
@@ -133,13 +133,26 @@ -(void) drawRect:(NSRect)rect
 
 -(void) noteKeyEvent:(KCKeystroke*)keystroke
 {
-	_displayedString = [[keystroke convertToString] retain];
+    if (_displayedString.length) {
+        [_displayedString appendString:[keystroke convertToString]];
+
+
+        if (_displayedString.length > 6) {
+            NSRange range = NSMakeRange(0, _displayedString.length - 6);
+            [_displayedString deleteCharactersInRange:range];
```

Comment:
...and [NSString substringWithRange:] here.  That way we avoid having to change the interface, which shouldn't be mutable anyway.